conv2d

rum1887 opened the pull request:
Add Layers to Paddle Backend #10696

rum1887 commented:
hi, @MahmoudAshraf97 kindly review this pr :)

MahmoudAshraf97 reviewed (changes requested):
paddle already has data format argument, why not use it directly instead of the axis swap?

rum1887 commented:
> paddle already has data format argument, why not use it directly instead of the axis swap?

yeah, so I initialled the data format to "NCHW" (PyTorch, Paddle has this format as the default whilst Tensorflow has "NHCW" as the default dataformat). 

I changed the code to perform axis swap once at the end.

works?

rum1887 commented:
> > paddle already has data format argument, why not use it directly instead of the axis swap?
> 
> Yes!
> 
> I initialized the data format to "NCHW". PyTorch, Paddle has this format as the default whilst Tensorflow has "NHCW" as the default dataformat.
> 
> I changed the code to perform axis swap once at the end.
> 
> works?

umm, since Paddle produces output consistent with the input format passed, I removed the axis swaps.

MahmoudAshraf97 commented on ivy/functional/backends/paddle/layers.py:
```diff
@@ -118,8 +118,22 @@ def conv2d(
-) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+):
+    df = "NHWC"
+    if data_format == "NCHW":
+        df = "NCHW"
```
I still don't see a need for this, what does this snipped add?

rum1887 replied:
yeah, its nit needed

rum1887 commented:
> please use the hypothesis tests to make sure your implementation is correct before requesting another review

yes, will do that!

why are the axes of filters switched in conv3d?
